api/client/swarm/init.go

```diff
@@ -36,7 +37,8 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.Var(&opts.listenAddr, "listen-addr", "Listen address")
+	flags.Var(&opts.listenAddr, flagListenAddr, "Listen address")
+	flags.StringVar(&opts.listenInterface, flagListenInterface, "", "Listen on an interface's address")
```

Comment:
I'm okay with adding `--listen-port` (as described in https://github.com/docker/docker/pull/24237#discussion_r69359620) if you're unhappy with putting the interface name and port in a single option. It's not my first choice because it feels a little clunky to use two options to configure what address the node binds to.

Any thoughts for or against this?